libafl_bolts/src/llmp.rs

```diff
@@ -1625,18 +1646,19 @@ where
                     self.last_msg_recvd = ptr::null();
                     self.highest_msg_id = MessageId(0);
 
-                    // Mark the old page save to unmap, in case we didn't do so earlier.
-                    (*page).safe_to_unmap.store(1, Ordering::Relaxed);
+                    // Mark the old page save to remap.
+                    (*page).receiver_left();
```

Comment:
shouldn't this be joined()
safe_to_unmap is
1 = receiver_joined()
0 = receiver_left()

Reply:
It's confusing that I'm reusing this line here, but I think it's correct. This happens when unmapping the page on receiver side.
- earlier we were just setting the page to 1 here (even though it was always already 1)
- Now we don't do that anymore since it would break proper refcounts if we ever introduce them

- Independently, we need to tell the sender that we're done with this page and it's safe to reuse it (`receiver_left`).
- This counts up an extra value and is what was missing before - hence the wraparound.
sounds correct?

Reply:
ok